### PR TITLE
Add qubit_map to SimulationTrialResult

### PR DESCRIPTION
**Pull request by mpharrigan**
xref https://github.com/quantumlib/Cirq/pull/1678#issuecomment-500966950

### CODE REVIEW

File: cirq/sim/simulator.py [hunk 1]
```diff
@@ -532,3 +532,7 @@ def _value_equality_values_(self):
         measurements = {k: v.tolist() for k, v in
                         sorted(self.measurements.items())}
         return (self.params, measurements, self._final_simulator_state)
+
+    @property
+    def qubit_map(self):
```
Comment (Strilanc) [inline cirq/sim/simulator.py]: Add a docstring explaining how this would be used, and also a return type annotation.

Comment (mpharrigan) [reply cirq/sim/simulator.py]: done

**Comment by mpharrigan** [comment]
ah, it would appear this breaks everything

**Comment by mpharrigan** [comment]
So `StateVectorMixin` -- which `WaveFunctionTrialResult` mixes in -- tries to set `qubit_map` which resolves to `SimulationTrialResult`'s property. hmm

**Comment by mpharrigan** [comment]
Ok, I may have fixed it. Please give it a look

**Comment by dabacon** [comment]
Looks like CirqBot is having issues.

**Pull request merged**